### ISSUE TITLE
📎 Website accessibility audit

### ISSUE THREAD

**Issue by sebmck** [rome/tools#839]
- Contrast
- Screenreader

**Comment by estevanmaito** [comment]
Do you have something specific in mind for this @sebmck ?

I usually ensure my projects are accessible using NVDA (screen reader) and a mix of browser tools and reports to catch contrast/sizing/labeling issues, etc. and I could help with this.

**Comment by sebmck** [comment]
I never went over the website with a screen reader. I think @EduardoLopes might have, but wanted to make sure that we weren't doing anything funky. Our docs are a single page so want to make sure the anchor hashes are sufficient enough to make navigating with a screen reader possible.

**Comment by EduardoLopes** [comment]
I did, but there was quite a lot changes since that and i couldn't do another pass yet. Your help will be much appreciated, @estevanmaito.

I tried to make sure contrast/sizing/labeling is good (we might have missed some labeling), but if you think we can improve on that, we can change it.

**Comment by ematipico** [comment]
As initial check, we could use [axe](https://www.deque.com/axe/). They have a chrome devtools that does an accessibility check on the website.

**Comment by estevanmaito** [comment]
Below you find a list of problems found in my audit for the home page.

# Screen Reader

## General

- [ ] The **Latest blog post** section at the top could be moved to the bottom of the file and reordered with CSS so it's not the first thing announced when navigating to the page.
- [ ] Header anchors (the `#` link after titles) don't have text, so titles are announced as (117 times) eg.:
  - Heading level 2, "Development Status", Heading Level 2, Link, "Number"
- [ ] Theme toggler should contain `role="button"` and a descriptive label

## Language Support, Table

Mostly problems with ✅, ⌛ and empty cells

- [ ] "Parsing, column 2, checkmark button" the check emoji is interpreted as a button and have no meaning
- [ ] "Linting, column 4, " column is empty, it's difficult to understand even seeing it. What does an empty cell mean?
- [ ] The ⌛ are inside `span`s with `aria-label="Work in progress"`, but are ignored as `span`s don't have semantic meaning, so they are read as: "Linting, column 4, hourglass done". A solution may be to add `role="img"` (and apply it to all emoji in the table)

## Large blocks of code in general
Specially those in [Linting](https://romefrontend.dev/#linting)
- [ ] The code examples are too big and detailed and could be hidden or announced like "Code snippet". The same applies to "[Suggested fixes](https://romefrontend.dev/#suggested-fixes)", "[Reviewing](https://romefrontend.dev/#reviewing)", [rome init](https://romefrontend.dev/#rome-init), [rome recover list](https://romefrontend.dev/#rome-recover-list)

# General a11y

- [ ] `.token.variable` (code syntax highlighting) have a contrast ratio of 4.38 and fails WCAG AA recommended 4.5
- [ ] Info messages in code (the ones with an ℹ) have a contrast ratio of 4.33
- [ ] Code examples with "Click to Expand" areas are not keyboard accessible
- [ ] Search input doesn't have a label.
- [ ] Sidebar content, top "Latest blog post" and main "Rome" logo + slogan are outside landmarks. Could be solved with some `aside`/`header`/`section` for example.
- [ ] Discord links have different `href` (triggers "Links with the same name have a similar purpose"). Footer link points to `https://discord.gg/9WxHa5d` (the same place as the top one, but that points to the pretty URL)

# Critical points

I would prioritize the fixes in this order:

1. Language support table: empty cells are confusing even for non-impaired readers and the ones with emojis are not correctly read by screen readers.
2. Header links:  annoying, meaningless and there are so much of them ~117
3. Announce/hide large code blocks (needs discussing, as everything, but this would be more _invasive_)
4. Keyboard inaccessible code blocks
5. The rest

Would love to hear your opinions!

**Comment by EduardoLopes** [comment]
Sweet! ✨ This looks good to me. What's you plan to fix the language support table and header links?

**Comment by estevanmaito** [comment]
1. Language support: I would like to understand first what the empty cells mean. Are they being worked or will never be developed? In the last case, I would add ⛔ and a meaningful label.

For context:

![Screenshot_2020-08-12 Rome Frontend Toolchain](https://user-images.githubusercontent.com/8527573/90028550-31600880-dc90-11ea-85fb-fb988b305fca.png)


2. Headings: duplicate the header content inside the link. I've seen approaches like add 'anchor' to the anchor (redundant in the sense that anchor is already announced and no context) or 'permalink' (complex and still no context). Duplicating the header is repetitive, but is clear and provides context.

**Comment by EduardoLopes** [comment]
For me, you can already start to work on it. i'll ping @sebmck here, he made the the table, he'll explain it better

**Comment by sebmck** [comment]
Ah yeah, the empty cells indicate that support isn't necesary. What JSON linting includes isn't obvious, although there's likely something we could do such as JSON schema validation, but nothing is planned right now. We can just fill them with check boxes though for consistency.

**Comment by estevanmaito** [comment]
Great. I'll start working on it on Monday.
